base/set.jl

```diff
@@ -101,8 +101,7 @@ function in!(x, s::Set)
 end
 
 push!(s::Set, x) = (s.dict[x] = nothing; s)
-pop!(s::Set, x) = (pop!(s.dict, x); x)
-pop!(s::Set, x, default) = (x in s ? pop!(s, x) : default)
```

Comment:
I believe, `pop!` should be optimized for `Set` to hash only once. The problem comes from the fact that there is no token API to implement `pop!` hashing only once through `AbstractSet` (#24454). I have added a similar optimization into internal `Base.in!` (#45156).

Reply:
That optimization is already present for `pop!(s::Set, x)`. This PR keeps it that way.
You may want `pop!(s::Set, x, default)` to also have that optimization, but that doesn't justify keeping around a redundant method that does not have the optimization.